amethyst_renderer/src/tex_animation.rs

```diff
@@ -36,6 +37,43 @@ impl Into<Result<SpriteSheetData, ()>> for SpriteSheetData {
     }
 }
 
+/// Error returned by the `set_animation` function on the `SpriteSheetAnimation`
+#[derive(Copy, Clone, Debug)]
+pub enum SetAnimationError {
```

Comment:
You can use `error-chain` for this, that would give us a back trace in case of an error.